src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -399,32 +400,49 @@ def open_resolve_dialog(self, slack_request: SlackActionRequest, group: Group) -
             callback_id["rule"] = slack_request.callback_data.get("rule")
         callback_id = orjson.dumps(callback_id).decode()
 
-        slack_client = SlackClient(integration_id=slack_request.integration.id)
-
         # XXX(CEO): the second you make a selection (without hitting Submit) it sends a slightly different request
         modal_payload = self.build_resolve_modal_payload(callback_id)
-        try:
-            payload = {
-                "view": orjson.dumps(modal_payload).decode(),
-                "trigger_id": slack_request.data["trigger_id"],
-            }
-            headers = {"content-type": "application/json; charset=utf-8"}
-            slack_client.post(
-                "/views.open",
-                data=orjson.dumps(payload).decode(),
-                headers=headers,
-            )
-        except ApiError as e:
-            logger.exception(
-                "slack.action.response-error",
-                extra={
-                    "error": str(e),
-                    "organization_id": org.id,
-                    "integration_id": slack_request.integration.id,
+        if features.has("organizations:slack-sdk-action-view-open", group.project.organization):
+            slack_client = SlackSdkClient(integration_id=slack_request.integration.id)
+            try:
+                slack_client.views_open(
+                    trigger_id=slack_request.data["trigger_id"],
+                    view=modal_payload,
```

Comment:
any chance we can make `modal_payload` into a `View` from `slack_sdk`? `views_open` converts the `View` back into a dict but it would be good to verify the payload